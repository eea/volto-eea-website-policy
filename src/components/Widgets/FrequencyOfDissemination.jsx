import React from 'react';
import cx from 'classnames';

export const FrequencyOfDissemination = ({ value, children, className }) => {
  if (!value) {
    return '';
  }

  const text = value > 1 ? `Every ${value} years` : 'Once a year';
  return (
    <span className={cx(className, 'text', 'widget')}>
      {children ? children(text) : text}
    </span>
  );
};
