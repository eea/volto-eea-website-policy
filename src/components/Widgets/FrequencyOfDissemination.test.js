import React from 'react';
import { render } from '@testing-library/react';
import { FrequencyOfDissemination } from './FrequencyOfDissemination';

describe('FrequencyOfDissemination', () => {
  it('renders "Once a year" when value is 1', () => {
    const { getByText } = render(<FrequencyOfDissemination value={1} />);
    expect(getByText('Once a year')).toMatchSnapshot();
  });

  it('renders "Every 2 years" when value is 2', () => {
    const { getByText } = render(<FrequencyOfDissemination value={2} />);
    expect(getByText('Every 2 years')).toMatchSnapshot();
  });

  it('renders custom text when children prop is provided', () => {
    const { getByText } = render(
      <FrequencyOfDissemination value={3}>
        {(text) => `Frequency: ${text}`}
      </FrequencyOfDissemination>,
    );
    expect(getByText('Frequency: Every 3 years')).toMatchSnapshot();
  });

  it('renders nothing when value is falsy', () => {
    const { container } = render(<FrequencyOfDissemination value={0} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
