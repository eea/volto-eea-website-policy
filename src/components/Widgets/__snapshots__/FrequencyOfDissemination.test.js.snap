// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrequencyOfDissemination renders "Every 2 years" when value is 2 1`] = `
<span
  class="text widget"
>
  Every 2 years
</span>
`;

exports[`FrequencyOfDissemination renders "Once a year" when value is 1 1`] = `
<span
  class="text widget"
>
  Once a year
</span>
`;

exports[`FrequencyOfDissemination renders custom text when children prop is provided 1`] = `
<span
  class="text widget"
>
  Frequency: Every 3 years
</span>
`;

exports[`FrequencyOfDissemination renders nothing when value is falsy 1`] = ``;
